File: articles/linear-algebra-for-systems-modeling/articles/pivot-structure-and-solvability/julia/pivot_structure_audit.jl
A = [
    1.0 1.0 0.0;
    0.0 1.0 1.0;
    1.0 0.0 1.0
]

b = [100.0, 80.0, 90.0]

rank_A = 3
rank_augmented = 3
unknown_count = size(A, 2)
consistent = rank_A == rank_augmented
solution_behavior = consistent && rank_A == unknown_count ? "unique solution" : "review needed"
pivot_columns = "0,1,2"
free_columns = "none"

println("system_name,equation_count,unknown_count,pivot_columns,free_columns,coefficient_rank,augmented_rank,consistent,solution_behavior,warning")
println(join((
    "three_constraint_resource_balance_system",
    size(A, 1),
    unknown_count,
    pivot_columns,
    free_columns,
    rank_A,
    rank_augmented,
    consistent,
    solution_behavior,
    "Pivot structure reveals algebraic solvability but feasibility requires review."
), ","))

File: articles/linear-algebra-for-systems-modeling/articles/pivot-structure-and-solvability/c/pivot_structure_audit.c
#include <stdio.h>

int main(void) {
    int equation_count = 3;
    int unknown_count = 3;
    int coefficient_rank = 3;
    int augmented_rank = 3;
    int consistent = 1;
    double tolerance = 1e-10;

    printf("system_name,equation_count,unknown_count,pivot_columns,free_columns,coefficient_rank,augmented_rank,consistent,solution_behavior,tolerance,warning\n");
    printf("three_constraint_resource_balance_system,%d,%d,0;1;2,none,%d,%d,%d,unique solution,%.10f,Pivot structure reveals algebraic solvability but feasibility requires review.\n",
           equation_count, unknown_count, coefficient_rank, augmented_rank, consistent, tolerance);
    return 0;
}
